### PR TITLE
Resolve wrapping in Firefox, added truncation in month select

### LINKED ISSUE #1243: [terra-date-input] Firefox wrapping the year input when set to inline
# Bug Report

## Description
Firefox displays the year input wrapped on the next line when DateInputField is set to `isInline`

## Steps to Reproduce
1. See the `Inline DateInputField` example in Firefox

## Additional Context / Screenshots
Screen Shot 2020-09-12 at 9 02 06 PM

## Expected Behavior
When set to Inline, should not wrap the year input.

## Possible Solution
https://github.com/cerner/terra-framework/blob/0cbfdf5745af49f9e3b4273a34253078e1dd9aed/packages/terra-date-input/src/DateInput.module.scss#L10
Revert to original default styles before PR #1230, `flex-wrap: wrap` to `flex-wrap: nowrap`

### PR DESCRIPTION
**Pull request by neilpfeiffer**
### Summary

Resolved the wrapping of the year input in Firefox when DateInputField was set to `isInline` by reverting to default style of `flex-wrap: nowrap` (as was prior to #1230).  Also added truncation to the month select dropdown in the case the width available to the field is less than standard needed, since wrapping is being prevented (otherwise the arrow overlaps the text).

Closes #1243

### Deployment Link

- Test the `Inline DateInputField` example on Firefox, validate the `YEAR` input doesn't drop to the next line:
https://terra-framew-date-input-8g9nja.herokuapp.com/components/terra-date-input/date-input/date-input-field

### Testing
Inline DateInputField no longer have year wrapping to next line in Firefox:
Screen Shot 2020-09-14 at 1 33 33 AM

When width is not wide enough to show full month,day,year group, the select will truncate its text:
Screen Shot 2020-09-14 at 1 34 21 AM

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-date-input/src/DateInput.module.scss [hunk 1]
```diff
@@ -126,12 +126,15 @@
...
-    margin: 0 2rem 0 0;
+    margin: 0 0.92857rem 0 0;
     outline: 0;
+    overflow: hidden;
     padding-bottom: var(--terra-date-input-month-select-padding-bottom, 0);
     padding-left: var(--terra-date-input-month-select-padding-left, 0.3571rem);
-    padding-right: var(--terra-date-input-month-select-padding-right, 0.3571rem);
+    padding-right: var(--terra-date-input-month-select-padding-right, 1.42857rem);
     padding-top: var(--terra-date-input-month-select-padding-top, 0);
+    text-overflow: ellipsis;
```
Comment (benbcai) [inline packages/terra-date-input/src/DateInput.module.scss:136]: 👍

**Review by benbcai — APPROVED** [review]
+1 with screenshot updates.

**Review by ShettyAkarsh — APPROVED** [review]

**Review by mjhenkes — APPROVED** [review]

**Comment by neilpfeiffer** [comment]
Screenshots updated here: https://github.com/cerner/terra-framework/pull/1245/commits/a5ccbaa896d52f73c9f25f5ccfbf004005aee5dd

**Comment by PaulSolDev** [comment]
Was unable to reproduce this issue on the existing component example. Sync'd with @neilpfeiffer  and noted wrapping issue on his device/browser and the corrected display with the new deployment. No further issues noted. Functional review complete.

**Review by emilyrohrbough — APPROVED** [review]

**Review by tbiethman — APPROVED** [review]

**Review by jeremyfuksa — APPROVED** [review]

**Pull request merged**